model/annotations/task_annotations.go

```diff
@@ -162,15 +162,15 @@ func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
 		a.Note.Source = source
 		update[NoteKey] = a.Note
 	}
-	if a.Issues != nil {
-		for _, issue := range a.Issues {
-			issue.Source = source
+	if len(a.Issues) != 0 {
```

Comment:
Oh, I actually don't think we should do this. If there's currently an Issue in issue, how else do I indicate I want to remove it and replace it with nothing? I assume if I give a document `{"task_id": 123, "issues": []} ` issues would be not nil but length 0, and suspected_issues would be nil. Is this not the case?

Reply:
If issues is not provided, it it not nil, it is an empty array, so checking for nil is not an option. I hear the concern though, I'll think about it.

Reply:
In your example, suspected_issues would be `<[]github.com/evergreen-ci/evergreen/model/annotations.IssueLink> (length: 0, cap: 0)`

Reply:
This seems like a failure of https://github.com/evergreen-ci/evergreen/blob/39b9a4171b64ba3afea10685c549e5f02b2dc197/rest/model/task_annotations.go#L145; you could change this to return nil if t is nil? Assuming t comes in as nil for suspected_ssues in that example.

Reply:
thanks!